pkg/ingester/ingester_v2_test.go

```diff
@@ -0,0 +1,288 @@
+package ingester
+
+import (
+	"io/ioutil"
+	"math"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/ring"
+	"github.com/cortexproject/cortex/pkg/util/test"
+	"github.com/cortexproject/cortex/pkg/util/validation"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/user"
+	"golang.org/x/net/context"
+)
+
+func Test_Ingester_v2LabelNames(t *testing.T) {
+	series := []struct {
+		lbls      labels.Labels
+		value     float64
+		timestamp int64
+	}{
+		{labels.Labels{{Name: labels.MetricName, Value: "test_1"}, {Name: "status", Value: "200"}, {Name: "route", Value: "get_user"}}, 1, 100000},
+		{labels.Labels{{Name: labels.MetricName, Value: "test_1"}, {Name: "status", Value: "500"}, {Name: "route", Value: "get_user"}}, 1, 110000},
+		{labels.Labels{{Name: labels.MetricName, Value: "test_2"}}, 2, 200000},
+	}
+
+	expected := []string{"__name__", "status", "route"}
```

Comment:
The result of `LabelNames()` is always sorted in Prometheus. If we were to follow that in Cortex too, then this should be fixed.

Reply:
Yes, but this is the ingester, not the querier, so ordering is not required at this stage. The querier fetch label names from all ingesters, and then sort it. Technically it's implemented in the distributor (which runs as internal dependency of the querier) and the [specific code is here](https://github.com/cortexproject/cortex/blob/master/pkg/distributor/distributor.go#L499).

_I'm going to resolve this conversation, but feel free to re-open it if something is missing._

Reply:
Ah got it!